cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -1068,7 +1082,7 @@ private Node getNode(Compute compute, Node.Type type, String zone, String nodeNa
       if (e.getStatusCode() == 404) {
         return new Node(nodeName, Node.Type.UNKNOWN, "", -1L, Collections.emptyMap());
       }
-      throw e;
+      throw new ErrorCodeWrappedException(e, ErrorGroup.DATAPROC, Integer.toString(e.getStatusCode()));
```

Comment:
is the error code supposed to be something human readable? It seems odd to have strings and http status codes. What about making this 'http-error-[statuscode]' to at least indicate it was a request failure.

That being said, we should be able to give more detailed information here based on the response. Is there documentation somewhere around what are the possible error codes and what they mean?

Reply:
Documentation is necessary and will be added. Created an issue to track https://cdap.atlassian.net/browse/CDAP-19410. http-status-errorcode is a good idea.

Reply:
Documentation meaning GCP docs on this API and the possible response codes, not CDAP docs. These errors should be as specific as possible. Using http status is just a fallback when we don't know how the API is expected to respond.